src/components/Dashboard/FilingHistoryList.vue

```diff
@@ -636,7 +646,13 @@ export default class FilingHistoryList extends Mixins(
     if (index >= 0) this.togglePanel(index, this.historyItems[index])
   }
 
-  private async correctThisFiling (item: HistoryItemIF): Promise<void> {
+  private startCorrectionDialog (item: HistoryItemIF): void {
+    this.fileCorrectionDialog = true
+    this.correctionFiling = item
```

Comment:
I recommend swapping these 2 lines so that the filing is set before the dialog is displayed.

(JS is single-threaded and run-to-completion, and these statements are atomic, so it probably works fine as-is but conceptually it makes more sense the other way IMO.)